source/builtin/src/lib.rs

```diff
@@ -272,6 +272,48 @@ impl<A> std::ops::Deref for Tracked<A> {
     }
 }
 
+macro_rules! emit_phantom {
+    ($x:ident) => {
+        PhantomData
+    };
+}
```

Comment:
what's this for?

Reply:
It's a hacky way to have the macro generate `$typ { phantom: PhantomData },` n times.  I'm not sure if there's a more natural way to do it.

Reply:
If I understand why you've introduced this bit of code correctly, you are introducing it as a way to "eat up" the `$x`, which comes from the `$t` in `split_tuple`, because otherwise Rust wouldn't know what your `$(...)*` repetition is set up for.

If that understanding is correct, then you should be able to eliminate this `emit_phantom` macro by replacing `emit_phantom!($t)` in `split_tuple` below with `PhantomData::<$t>`. That way, you still have the `$t` anchoring the repetition.

Since the solution is to use a turbofish, please enjoy this link: [`::<$t>`](https://turbo.fish/::%3C$t%3E) 😄